src/de/error.rs

```diff
@@ -15,6 +15,12 @@ pub enum Error {
     /// while decoding a UTF-8 String from the input data.
     InvalidUtf8String(string::FromUtf8Error),
 
+    /// An error encountered during the conversion of one datetime format to another.
+    InvalidTimestamp {
```

Comment:
rather than reusing the deserialization error here, I wonder if it would be better to introduce another error type for datetimes like we have for ObjectIds (`oid::Error`, returned from `ObjectId::parse_str`) and UUIDs (`uuid::Error`, returned from `Uuid::parse_str`). It's unfortunate that this crate has so many error types, but I think it's something we probably have to commit to at this point.

cc @abr-egn @isabelatkinson

Reply:
Yeah that seems fine to me for the sake of consistency

Reply:
Added an error enum and switched over to using that instead in https://github.com/mongodb/bson-rust/pull/317/commits/0bd1b009c5d1a4bad0760a9a68bdc66936f77845.